### PR TITLE
Implement CWLogs log group class

### PR DESCRIPTION
**Pull request by movence**
# Description of changes
- add `log_group_class` flag and validation for it

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Tested on EC2 instance
Agent config used:
```
{
  "agent": {
    "run_as_user": "root",
    "debug": true
  },
  "logs": {
    "logs_collected": {
      "files": {
        "collect_list": [
          {
            "file_path": "/var/essentials.log",
            "log_group_name": "{instance_id}-essentials",
            "log_stream_name": "{instance_id}",
            "timezone": "UTC",
            "log_group_class": "infrequent_access"
          },
          {
            "file_path": "/var/standard.log",
            "log_group_name": "{instance_id}-standard",
            "log_stream_name": "{instance_id}",
            "timezone": "UTC",
            "log_group_class": "standard"
          }
        ]
      }
    }
  }
}
```

Screenshot 2023-11-29 at 2 09 54 PM

# Requirements

_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

**Review by dchappa — APPROVED** [review]

File: tool/data/config/logs/eventConfig.go [hunk 1]
```diff
@@ -25,6 +26,9 @@ func (config *EventConfig) ToMap(ctx *runtime.Context) (string, map[string]inter
 	}
 	resultMap["log_group_name"] = config.LogGroup
 	resultMap["log_stream_name"] = config.LogStream
+	if config.LogGroupClass != "" {
+		resultMap["log_group_class"] = config.LogGroupClass
```
Comment (lisguo) [inline tool/data/config/logs/eventConfig.go:30]: is there any validation on the log_group_class configuration? ie. check for STANDARD, INFREQUENT_ACCESS, etc?

File: translator/translate/logs/util/validate_fields.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+func ValidateLogGroupFields(logConfigs []interface{}, currPath string) []interface{} {
+	logConfigs = validateLogRetentionSettings(logConfigs, currPath)
+	logConfigs = validateLogGroupClassSettings(logConfigs, currPath)
+	return logConfigs
+}
+
+func validateLogRetentionSettings(logConfigs []interface{}, currPath string) []interface{} {
+	configMap := make(map[string]int)
+	for _, logConfig := range logConfigs {
+		if logConfigMap, ok := logConfig.(map[string]interface{}); ok {
```
Comment (lisguo) [inline translator/translate/logs/util/validate_fields.go:30]: nit: I would prefer less nesting by checking for the negative case first:

```
if logConfigMap, ok := logConfig.(map[string]interface{}); !ok {
    continue 
}
if retention, ok := logConfigMap[logRetentionKey].(int); !ok {
    continue
}
 ...
 ```

File: translator/translate/logs/util/validate_fields.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+func validateLogGroupClassSettings(logConfigs []interface{}, currPath string) []interface{} {
+	configMap := make(map[string]string)
+	for _, logConfig := range logConfigs {
+		if logConfigMap, ok := logConfig.(map[string]interface{}); ok {
+			// skip if logGroupClass is not set
+			if logGroupClass, ok := logConfigMap[logGroupClassKey].(string); ok {
+				// if logGroupClass is not one of the proper values, it's invalid skip it
+				if !slices.Contains(translator.ValidLogGroupClasses, logGroupClass) {
```
Comment (lisguo) [inline translator/translate/logs/util/validate_fields.go:64]: I don't see `ValidLogGroupClasses` slice anywhere in this PR. Does it already exist?

**Comment by lisguo** [comment]
nit: PR title is a bit misleading. This is the implementation of CWLogs log group classes. Not only infrequent access

File: .github/repo_sync_pr_template.md
Comment (lisguo) [inline .github/repo_sync_pr_template.md:1]: is this file needed? Looks like it's empty

File: translator/translate/logs/util/validate_fields.go (hunk 3 above)
Comment (movence) [reply translator/translate/logs/util/validate_fields.go:64]: https://github.com/aws/amazon-cloudwatch-agent/pull/968/files#diff-dc5b0b91cd83446482896b1d49047f1944559f05025900d5ec6e22d32698862eR22

File: .github/repo_sync_pr_template.md
Comment (movence) [reply .github/repo_sync_pr_template.md:1]: probably not. I will remove it

File: tool/data/config/logs/eventConfig.go (hunk 1 above)
Comment (dchappa) [reply tool/data/config/logs/eventConfig.go:30]: There are validations to make sure that the only STANDARD and INFREQUENT_ACCESS are accepted https://github.com/aws/amazon-cloudwatch-agent/pull/968/files#diff-168367fbe6308e4f71d0d767b73db774d4a04414627718695c13c7d5709b2e05R57

File: translator/translate/logs/util/validate_fields.go (hunk 2 above)
Comment (dchappa) [reply translator/translate/logs/util/validate_fields.go:30]: Can follow up in a separate PR? This was moved from a different file to simplify logic, not related to this PR

**Review by dchappa — APPROVED** [review]

**Review by lisguo — APPROVED** [review]
